src/server/scripts/EasternKingdoms/Deadmines/boss_vancleef.cpp

```diff
@@ -0,0 +1,176 @@
+};
+
+class boss_vancleef : public CreatureScript
+{
+public:
+    boss_vancleef() : CreatureScript("boss_vancleef") { }
+
+    struct boss_vancleefAI: public BossAI
+    {
+        boss_vancleefAI(Creature* creature) : BossAI(creature, 0)
+        {
+            _phase = PHASE_NORMAL;
+        }
+
+        void Initialize()
+        {
+            DoCastSelf(SPELL_DUAL_WIELD, true);
+            DoCastSelf(SPELL_THRASH, true);
+
+            _SummonBlackguards();
+        }
+
+        void EnterCombat(Unit* victim) override
+        {
+            BossAI::EnterCombat(victim);
+            summons.DoZoneInCombat();
+
+            Talk(SAY_AGGRO);
+        }
+
+        void KilledUnit(Unit* victim) override
+        {
+            if (victim->GetTypeId() == TYPEID_PLAYER)
+                Talk(SAY_KILL);
+        }
+
+        void EnterEvadeMode(EvadeReason /*why*/) override
+        {
+            summons.DespawnAll();
+            _DespawnAtEvade();
+        }
+
+        void Reset() override
+        {
+            BossAI::Reset();
+            Initialize();
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            scheduler.Update(diff, [this]
+            {
+                DoMeleeAttackIfReady();
+            });
+        }
+
+        void _SummonBlackguards()
+        {
+            for (auto BlackguardPosition : BlackguardPositions)
+                DoSummon(NPC_BLACKGUARD, BlackguardPosition, 60000,
+                        TEMPSUMMON_CORPSE_TIMED_DESPAWN);
+        }
+
+        void _SwitchPhase(uint8 Phase)
+        {
+            // don't repeat
+            if (_phase == Phase)
+                return;
+
+            _phase = Phase;
+
+            switch (_phase)
+            {
+            case PHASE_66_PCT:
+                Talk(SAY_ONE);
+                break;
+            case PHASE_50_PCT:
+                Talk(SAY_SUMMON);
+                DoCastSelf(SPELL_VANCLEEFS_ALLIES);
+                break;
+            case PHASE_33_PCT:
+                Talk(SAY_TWO);
+                break;
+            case PHASE_25_PCT:
+                Talk(SAY_THREE);
+                break;
+            default:
+                break;
+            }
+        }
+
+        void ScheduleTasks() override
+        {
+            scheduler.Schedule(Seconds(1), [this](TaskContext task)
+            {
+                if (_phase == PHASE_NORMAL && HealthBelowPct(66))
+                    _SwitchPhase(PHASE_66_PCT);
+                else if (_phase == PHASE_66_PCT && HealthBelowPct(50))
+                    _SwitchPhase(PHASE_50_PCT);
+                else if (_phase == PHASE_50_PCT && HealthBelowPct(33))
+                    _SwitchPhase(PHASE_33_PCT);
+                else if (_phase == PHASE_33_PCT && HealthBelowPct(25))
+                    _SwitchPhase(PHASE_25_PCT);
+
+                task.Repeat(Seconds(1));
+            });
+        }
+
+    protected:
```

Comment:
private

Reply:
Changed to all your suggestions, plus used the new model.